src/librustc_traits/dropck_outlives.rs

```diff
@@ -227,15 +227,16 @@ fn dtorck_constraint_for_ty<'tcx>(
             // In particular, skipping over `_interior` is safe
             // because any side-effects from dropping `_interior` can
             // only take place through references with lifetimes
-            // derived from lifetimes attached to the upvars, and we
-            // *do* incorporate the upvars here.
+            // derived from lifetimes attached to the upvars and resume
+            // argument, and we *do* incorporate those here.
 
             constraints.outlives.extend(
                 substs
                     .as_generator()
                     .upvar_tys(def_id, tcx)
                     .map(|t| -> ty::subst::GenericArg<'tcx> { t.into() }),
             );
+            constraints.outlives.push(substs.as_generator().resume_ty(def_id, tcx).into());
```

Comment:
This is wrong. The resume argument type is not a type stored in the generator in the same way that argument types for closures are not stored in the closures themselves.

Reply:
Wouldn't that mean we'd have to consider the entire generator witness instead (since any type contained within may be stored in the generator across a `yield`)? AFAICT this is only dangerous if the types saved across a `yield` contain anything passed in from the outside (so upvars or the resume argument), right?

For closures, the only storage they can possibly have *is* their upvars, so there it works out.

Reply:
Yeah, I guess we'll need the resume argument here. Extra types should be a conservative option anyway.